subcommand/consul-init/command.go

```diff
@@ -0,0 +1,162 @@
+package consulinit
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/hashicorp/consul-k8s/consul"
+	"github.com/hashicorp/consul-k8s/subcommand/common"
+	"github.com/hashicorp/consul-k8s/subcommand/flags"
+	"github.com/hashicorp/consul/api"
+	"github.com/mitchellh/cli"
+)
+
+const bearerTokenFile = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+const tokenSinkFile = "/consul/connect-inject/acl-token"
+
+type Command struct {
+	UI cli.Ui
+
+	flagACLAuthMethod   string // Auth Method to use for ACLs, if enabled
+	flagMeta            string // Flag for metadata to consul login.
```

Comment:
So this field is used exclusively by `consul login` and is hard-coded by the init_container.
I think I can probably rename it to "-consul-login-meta" or honestly remove it altogether because at runtime we will actually know how to build up the string "pod=ns/podname".
Do we expect to be able to pass in additional metadata to `consul login` or do you envision us having other metadata to be passed into this command? Right now it does not support it, but I do plan to expand the command with additional flags so that it can do the rest of the init code!
Should I remove / rename this field?

Reply:
If we do ever need to add additional metadata to consul login in the future it'll probably be nice to have the []string. I think it'd be ok for us to assume the order of the flags and things like that since it'll always be passed programmatically and consistently for all connect-injected pods.